BUILD.bazel

```diff
@@ -45,6 +45,15 @@ clippy_flags(
     visibility = ["//visibility:public"],
 )
 
+# This setting may be used to identify dependencies of proc-macro-s.
+# Its value controls the BAZEL_RULES_RUST_IS_PROC_MACRO_DEP environment variable
+# made available to the rustc invocation.
+is_proc_macro_dep(
```

Comment:
I'd add a comment that one should not depend on this flag or the environment variable, in case having the transition as a default proves to be too expensive and we need to add an additional `is_bootstrap` flag that will control whether this flag is relevant or not.

Reply:
After chatting f2f, went ahead and added the new `is_proc_macro_dep_enabled` knob to control whether this feature is enabled (resulting in a no-op transition if is_proc_macro_dep_enabled is false, which is the default.